collector/src/main/java/io/prometheus/jmx/JmxCollector.java

```diff
@@ -398,6 +407,10 @@ public void recordBean(
       JmxScraper scraper = new JmxScraper(config.jmxUrl, config.username, config.password, config.ssl, config.whitelistObjectNames, config.blacklistObjectNames, receiver);
       long start = System.nanoTime();
       double error = 0;
+      if ((config.startDelaySeconds > 0) &&
+        ((start - createTimeNanoSecs) / 1000000000L < config.startDelaySeconds)) {
+        throw new IllegalStateException("JVM not ready");
```

Comment:
This message might confuse the user about the JVM generally. I'd go for something like "JMXCollector waiting for startDelaySeconds "